solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -1270,7 +1270,7 @@ private void doProcessGroupedDistributedSearchFirstPhase(ResponseBuilder rb, Que
       final int topNGroups;
       Query query = cmd.getQuery();
       if (query instanceof AbstractReRankQuery){
-        topNGroups = cmd.getOffset() + ((AbstractReRankQuery)query).getReRankDocs();
+        topNGroups = Math.max(((AbstractReRankQuery)query).getReRankDocs(), cmd.getOffset() + cmd.getLen());
```

Comment:
Should this be handled for non distributed case as well in doProcessGroupedSearch(..), when pageSize>reRankDocs it returns only "reRankDocs" groups in non distributed case.

Reply:
the non distributed case looks fine to me.

Reply:
You were right, I added a unit test and it is failing :/ I'm working on it